### PR TITLE
Add time components to location place holders

### PR DESCRIPTION
**Pull request by bstdenis**
Write facets given as default to the NetCDF file.

### CODE REVIEW

File: malleefowl/persist.py [hunk 1]
```diff
@@ -8,38 +9,67 @@
...
+                facet_values['month'] = facet_values['initial_month']
+                if facet_values['initial_day'] == facet_values['final_day']:
+                    facet_values['day'] = facet_values['initial_day']
+
+    for facet in place_holders:
+        if facet in nc.ncattrs():
+            facet_values[facet] = nc.getncattr(facet)
+        elif "{0}_id".format(facet) in nc.ncattrs():
+            facet_values[facet] = nc.getncattr("{0}_id".format(facet))
+        elif facet in defaults:
```
Comment (dbyrns) [inline malleefowl/persist.py:51]: should be  "elif facet in facet_values:" no?

Comment (bstdenis) [reply malleefowl/persist.py:51]: no, at that point I don't want to add the time components to the netcdf global attributes, this is redundant, I just want to add the stuff in defaults that is not found in the ncattrs

File: malleefowl/persist.py [hunk 2]
```diff
@@ -8,38 +9,67 @@
...
+        facet_values['final_day'] = str(datetime_f.day).zfill(2)
+        # Could also add support for hour, minute, second, etc.
+
+        if facet_values['initial_year'] == facet_values['final_year']:
+            facet_values['year'] = facet_values['initial_year']
+            if facet_values['initial_month'] == facet_values['final_month']:
+                facet_values['month'] = facet_values['initial_month']
+                if facet_values['initial_day'] == facet_values['final_day']:
+                    facet_values['day'] = facet_values['initial_day']
+
```
Comment (dbyrns) [inline malleefowl/persist.py]: Should I understand that year/month/day will not be available unless initial and final values are the same?

Comment (bstdenis) [reply malleefowl/persist.py]: yes, otherwise what should the year value be?

File: malleefowl/persist.py (hunk 1 above)
Comment (dbyrns) [reply malleefowl/persist.py:51]: Ok I see the first 2 ifs work as extraction and the third one is populating the metadata from the defaults dict !

File: malleefowl/persist.py (hunk 2 above)
Comment (dbyrns) [reply malleefowl/persist.py]: And what is the point to add a third variable if 3 of them have the same value ? I mean if we cannot garantee that this facet is always available anyway.

File: malleefowl/persist.py [hunk 3]
```diff
@@ -8,38 +9,67 @@
 LOGGER = logging.getLogger("PYWPS")
 
 
-def resolve(location, place_holders, f):
+def resolve(location, f, defaults=None):
+    place_holders = re.findall('\{(.*?)\}', location)
+    if not place_holders:
+        return location
```
Comment (dbyrns) [inline malleefowl/persist.py]: In that case we miss the opportunity to add defaults value to the dataset metadata, right?

File: malleefowl/persist.py (hunk 2 above)
Comment (bstdenis) [reply malleefowl/persist.py]: good point, removing...

File: malleefowl/persist.py (hunk 3 above)
Comment (bstdenis) [reply malleefowl/persist.py]: right, fixing this...

File: malleefowl/persist.py [hunk 4]
```diff
@@ -8,38 +9,62 @@
...
+        facet_values['final_day'] = str(datetime_f.day).zfill(2)
+        # Could also add support for hour, minute, second, etc.
+
+    for facet in place_holders:
+        if facet in nc.ncattrs():
+            facet_values[facet] = nc.getncattr(facet)
+        elif "{0}_id".format(facet) in nc.ncattrs():
+            facet_values[facet] = nc.getncattr("{0}_id".format(facet))
+        elif facet in defaults:
+            nc.setncattr(facet, facet_values[facet])
```
Comment (dbyrns) [inline malleefowl/persist.py:52]: The problem with this step inside the loop is that we are looping on the place holders but this step has nothing to do with place holders... We should be looping over all the defauts facet (not just the one referenced in the place holders).

**Pull request merged**